### PR TITLE
feat: replace `micro-aes-gcm` with `@noble/chipers` [SPMVP-6208]

### PR DESCRIPTION
**Pull request by arielllin**
## Jira
https://storipress-media.atlassian.net/browse/SPMVP-6208

[//]: # 'This template is for new features, or changing an existing feature'
[//]: # 'Put the related jira links here'
[//]: # 'Please ensure there has description for the feature'

## Purpose
`micro-aes-gcm` 已經停止維護，更換為 `@noble/chipers`，並確認付費文章有正常加解密

[//]: # 'Please describe how the feature will affect user, e.g.'
[//]: # '- Allow to add link on image so publisher can directly link an image in static site'
[//]: # '- Adjust "Go to site" button to become "Go to Shopify" to let publisher using Shopify integration can directly go to see preview'
[//]: # (User should be one of "reader" {who read publisher's articles}, "publisher" {our users}, "developer" {us})

### For who

- [x] reader-facing?
- [ ] publisher-facing?
- [x] developer-facing?

## How did you make it?

[//]: # 'Give a brief for the changes you made'

## Production deployment notes

[//]: # 'Please describe the production deployment notes, e.g. this changes depend on other API or module change'

## 🎩 Tophat

Do a thorough 🎩. What is [tophatting](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md_)?

Consider testing:

- Existing functionality which could break due to your changes (e.g. global changes)
- New functionality being introduced. Ensure it meets intended behavior
- All different permutations (flows, states, conditions, etc) that are possible
- If you are modifying something which is used in multiple places, 🎩 all affected areas not just what you intend to change

### 🎩 Instructions

[//]: # "If it's complex, consider put a screen record here"

測試使用 `D6RX98VXN` 文章id `381` 這篇文章

1. 清除 local storage `storipress-token` 
2. 確認付費文章未顯示 ⭕ 
3. 增加 local storage `storipress-token` 模擬已登入狀態
4. 確認付費文章有顯示 ⭕ 

## Related PRs

[//]: # 'Put the related PRs here, e.g. PR in core-component'

## Checklist before requesting review

- [ ] I have done a self-review of my own code (comment on code is not required but recommended)
- [ ] I confirmed that I added review priority label (`priority:high`, `priority:medium`, ...)
- [ ] I did the Tophat steps and confirm the feature is working
- [ ] I have confirmed there is no issue reported by Static Analyzer (Please double check for custom class. For other issues, if you think it's ignorable, please add `// eslint-ignore-next-line <rule name>` on it)
- [ ] I have confirmed there is no deepsource issue (if you think it's ignorable, please explain why and add a [`skipcq` comment](https://deepsource.io/blog/releases-silence-issues-in-code/))
- [ ] I confirmed that the unit test is passed (if you break it, please fix it in this PR)

## Emoji Guide

**For reviewers: Emojis can be added to comments to call out blocking versus non-blocking feedback.**

E.g: Praise, minor suggestions, or clarifying questions that don’t block merging the PR.

> 🟢 Nice refactor!

> 🟡 Why was the default value removed?

E.g: Blocking feedback must be addressed before merging.

> 🔴 This change will break something important

|              |                |                                     |
| ------------ | -------------- | ----------------------------------- |
| Blocking     | 🔴 ❌ 🚨       | RED                                 |
| Non-blocking | 🟡 💡 🤔 💭    | Yellow, thinking, etc               |
| Praise       | 🟢 💚 😍 👍 🙌 | Green, hearts, positive emojis, etc |

## Gif (optional)

image

### CODE REVIEW

File: packages/karbon/src/runtime/composables/storipress-base-client.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ export function getStoripressConfig(): ModuleRuntimeConfig['storipress'] {
   try {
     const { storipress } = useRuntimeConfig()
     // @ts-expect-error type error
-    return storipress
+    return storipress ?? storipressConfigCtx.use()
```
Comment (arielllin) [inline packages/karbon/src/runtime/composables/storipress-base-client.ts:67]: 這邊是修正取 storipress config 會 undefined 的問題

截圖 2024-02-01 上午10 42 11

File: packages/karbon/src/runtime/api/normalize-article.ts [hunk 2]
```diff
@@ -73,6 +73,7 @@ export interface RawArticleLike {
 export interface PaidContent {
   content: string
   key: string
+  iv: number[]
```
Comment (DanSnow) [inline packages/karbon/src/runtime/api/normalize-article.ts]: 這邊可以同樣做 base64 encode 與 decode 嗎？

File: packages/karbon/src/runtime/api/article.ts [hunk 3]
```diff
@@ -347,17 +348,22 @@ async function encryptArticle({ plan, html, id, ...rest }: _NormalizeArticle) {
...
+    const content = await cipher.encrypt(encoder.encode(paid))
+
     const compactEncrypter = new CompactEncrypt(
       Buffer.from(JSON.stringify({ id, plan, key: Buffer.from(key).toString('base64') })),
     ).setProtectedHeader({ enc: 'A256GCM', alg: 'dir' })
     const encryptedKey = await compactEncrypter.encrypt(Buffer.from(storipress.encryptKey, 'base64'))
     paidContent = {
       key: encryptedKey,
       content: Buffer.from(content).toString('base64'),
+      iv: Array.from(iv),
```
Comment (DanSnow) [inline packages/karbon/src/runtime/api/article.ts]: 可以參考 content 的方式做 base64 encode

**Review by DanSnow — CHANGES_REQUESTED** [review]

File: packages/karbon/src/runtime/api/normalize-article.ts (hunk 2 above)
Comment (arielllin) [reply packages/karbon/src/runtime/api/normalize-article.ts]: 調整好了

File: packages/karbon/src/runtime/api/article.ts (hunk 3 above)
Comment (arielllin) [reply packages/karbon/src/runtime/api/article.ts]: 調整好了

**Review by DanSnow — APPROVED** [review]

**Pull request merged**